spec/factories/course_assessment_categories.rb

```diff
@@ -2,7 +2,7 @@
   sequence(:course_assessment_category_weight) { |n| n }
   sequence(:course_assessment_category_title) { |n| "Category #{n}" }
   factory :course_assessment_category, class: Course::Assessment::Category do
-    course
+    course { create(:course) }
```

Comment:
Maybe we should build the folder on validation, like other models, as part of the component callbacks?

Maybe the component callbacks should allowing hooking into the normal AR lifecycle?